tests/jenkins/TestPublishSmokeTestResults.groovy

```
@@ -0,0 +1,257 @@
+                smoke_test_build_url: 'http://example.com/build/123',
+                distribution_build_number: 456,
+                distribution_build_url: 'http://example.com/distribution/456',
+                // Ignore build_start_time for comparison
+                rc: 'rc1',
+                rc_number: 1,
+                platform: 'linux',
+                architecture: 'x64',
+                distribution: 'tar',
+                component_category: 'test-category',
+                test_report_manifest_yml: 'http://example.com/test-report.yml',
+                component_test_spec_name: 'GET__test_api',
+                component_test_spec_yml: 'test-spec-yml1',
+                component_test_spec_result: 'PASS',
+                component_test_spec_cluster_stdout: ['cluster_stdout1'],
+                component_test_spec_cluster_stderr: ['cluster_stderr1'],
+                "component_test_spec_test_stdout": ['test_stdout1'],
+                component_test_spec_test_stderr: ['test_stderr1'],
+        ]
+
+        // Remove the dynamic field for comparison
+        parsedResult.remove('build_start_time')
+        assert parsedResult == expectedJson
+    }
+
+    @Test
+    void testCallWithMissingArgs() {
+        def script = loadScript('vars/publishSmokeTestResults.groovy')
+        def args = [
+            distributionBuildUrl: "http://example.com/distribution/456",
+            testReportManifestYml: "path/to/testReportManifest.yml",
+        ]
+
+        def result = script.call(args)
+
+        assert result == null
+    }
+
+    @Test
+    void testCallWithEmptyArgs() {
+        def script = loadScript('vars/publishSmokeTestResults.groovy')
+        def args = [
+            distributionBuildUrl: "http://example.com/distribution/456",
+            testReportManifestYml: "path/to/testReportManifest.yml",
+            jobName: ""
+        ]
+
+        def result = script.call(args)
+
+        assert result == null
+    }
```

These should throw an error right? Should hard fail the job.